plugins/commands/winrm_config/command.rb

```diff
@@ -0,0 +1,124 @@
+require 'optparse'
+require "vagrant/util/safe_puts"
+
+module VagrantPlugins
+  module CommandWinRMConfig
+    class Command < Vagrant.plugin("2", :command)
+      include Vagrant::Util::SafePuts
+
+      def self.synopsis
+        "outputs WinRM configuration to connect to the machine"
+      end
+
+      def convert_win_paths(paths)
+        paths.map! { |path| Vagrant::Util::Platform.format_windows_path(path, :disable_unc) }
+      end
+
+      def execute
+        options = {}
+
+        opts = OptionParser.new do |o|
+          o.banner = "Usage: vagrant winrm-config [options] [name|id]"
+          o.separator ""
+          o.separator "Options:"
+          o.separator ""
+
+          o.on("--host NAME", "Name the host for the config") do |h|
+            options[:host] = h
+          end
```

Comment:
Hmm, this isn't clear to me how it's different from the CLI argument you can pass in or when this should be used over the CLI argument.